src/main.rs

```diff
@@ -1,4 +1,146 @@
+abigen!(ReserveAuctionCoreETH, "out/ListBidEssay.s.sol/ReserveAuctionCoreETH.json");
+abigen!(ModuleManager, "out/ListBidEssay.s.sol/ModuleManager.json");
+
+
+
+#[tokio::main]
+async fn main() {
+    let fork_endpoint = env::var("ETH_NODE_URL").expect("Environment variable ETH_NODE_URL should be defined and be a valid API URL");
+    let anvil = Anvil::new()
+        .fork(fork_endpoint)
+        .spawn();
+    let endpoint = anvil.endpoint();
+    println!("Anvil running at `{}`", endpoint);
+
+    // Set up anvil
+    let provider = Provider::<Http>::try_from(anvil.endpoint()).expect("Failed to connect to Anvil").interval(Duration::from_millis(10u64));
+
+    let wallet: LocalWallet = anvil.keys()[0].clone().into();
+    let multisig = wallet.address();
+    let client = Arc::new(SignerMiddleware::new(provider, wallet.with_chain_id(anvil.chain_id())));
+    let block_provider = Provider::<Http>::try_from(anvil.endpoint()).expect("Failed to connect to Anvil").interval(Duration::from_millis(10u64));
+
+    let block = block_provider.get_block_number().await.expect("Couldn't get block height");
+    let block_uint256 = ethers::prelude::U256::from(block.as_u64());
+    println!("Forked from block {}", block);
+
+    // Deploy contract
+    let nft_contract = task::block_on(SevenTeenTwentyNineEssay::deploy(client, multisig).expect("Failed to deploy").send()).expect("Failed to send");
+
+    // Mint Essay NFT
+    let mint_call = nft_contract.mint(multisig, "https://test.com/test".to_string());
+    task::block_on(mint_call.send()).expect("Failed to send mint transaction");
+
+    // Query Total Supply
+    let total_supply = nft_contract.method::<_, Uint>("totalSupply", ())
+        .expect("Error finding method").call().await.expect("Error sending total supply call");
+    println!("Total supply: {}", total_supply);
+
+    // Query royalty info
+    let token_id = ethers::core::types::U256::from_dec_str("1").expect("Couldn't convert 1 to U256");
+    let sale_price = ethers::core::types::U256::from_dec_str("5000000000000000000").expect("Couldn't convert 5*10^18 to U256");
+    let royalty_info = nft_contract.method::<_, (ethers::core::types::Address, ethers::core::types::U256)>("royaltyInfo", (token_id, sale_price))
+        .expect("Error finding royaltyInfo method").call().await.expect("Error sending royaltyInfo call");
+
+
+    println!("Royalty info: address: {}, amount: {}", royalty_info.0, royalty_info.1);
+
+    // List
+    let provider = Provider::<Http>::try_from(anvil.endpoint()).expect("Failed to connect to Anvil").interval(Duration::from_millis(10u64));
+    let wallet: LocalWallet = anvil.keys()[0].clone().into();
+    let client = Arc::new(SignerMiddleware::new(provider, wallet.with_chain_id(anvil.chain_id())));
+    let zora_address: ethers::core::types::Address = "0x5f7072E1fA7c01dfAc7Cf54289621AFAaD2184d0".parse::<Address>().expect("Couldn't parse zora address");
```

Comment:
Not blocking, iirc you mentioned we could come up w/ a better pattern for managing already-deployed contract addresses across various chains.